static/app/components/searchQueryBuilder/tokenizedQueryGrid.tsx

```diff
@@ -37,10 +37,15 @@ function useApplyFocusOverride(state: ListState<ParseResultToken>) {
   useLayoutEffect(() => {
     if (focusOverride && !focusOverride.part) {
       state.selectionManager.setFocused(true);
-      state.selectionManager.setFocusedKey(focusOverride.itemKey);
+
+      if (focusOverride.itemKey === 'end') {
```

Comment:
This adds support for putting the focus at the end of the query after a state update, which is what we need to do when selecting a recent search